core/libp2p/basic/writable.hpp

```diff
@@ -8,46 +8,27 @@
 
 #include <functional>
 
-#include <outcome/outcome.hpp>
-
-#include "common/buffer.hpp"
+#include <boost/asio/buffer.hpp>
```

Comment:
boost asio should be internal dependency. We can for example decide to use some another library to implement some protocol (i.e. introduce quic, which is not part of asio). Can we avoid using boost in interfaces?

Reply:
boost asio will be used anyway. We will just implement [Protocol requirements](https://www.boost.org/doc/libs/1_69_0/doc/html/boost_asio/reference/Protocol.html) and integrate quic in boost asio.